### PR TITLE
JP-2984: Update ASN rules for NIS_WFSS to keep grisms separate

### LINKED ISSUE #7315: Create separate spec3 ASN's for each WFSS grism orientation
_Issue [JP-2984](https://jira.stsci.edu/browse/JP-2984) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

Right now the asn_generator is creating spec3 ASN's that contain data from both (orthogonal) grism orientations when a WFSS observation contains exposures using both grisms (grismr and grismc for NIRCam, gr150r and gr150c for NIRISS). This results in level 3 products that contain combined data from both grisms. The INS teams wish to have the data from the different grism orientations kept separate all the way through Level 3 processing, hence resulting in 2 products for each source at each processing stage (e.g. 2 "cal", 2 "x1d", and 2 "c1d" products).

Therefore the ASN generator rules need to be updated to create separate ASN's for each grism contained in an observation. Instead of a single spec3 ASN with product names like "jw01085-o012_\{source_id}*niriss_f150w-gr150c-gr150r", there should be 2 ASN's, each with product names of "jw01085-o012*\{source_id}*niriss_f150w-gr150c" and "jw01085-o012*\{source_id}_niriss_f150w-gr150r" and only containing input exposures using that particular grism.

### PR DESCRIPTION
**Pull request by hbushouse**
Resolves [JP-2984](https://jira.stsci.edu/browse/JP-2984)

Closes #7315 

This PR makes a trivial change to the NIRISS WFSS level-3 ASN rule so that unique ASN's and products are created for each of the gr150r and gr150c grisms.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by hbushouse** [comment]
Regression test results are here: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/463/

2 failures (nirspec_ifu) are unrelated. The other 4 are all as expected, where formerly it was creating product names with optical elements of "-gr150c-gr150r" and now there are separate "-gr150c" and "-gr150r" products.

**Comment by hbushouse** [comment]
CI failures are the usual random glitches in file access.

**Review by tapastro — APPROVED** [review]
Looks good!

File: CHANGES.rst [hunk 1]
```diff
@@ -19,6 +19,8 @@ associations
 
 - Suppress the use of association candidates for level 3 products marked with WFSC_LOS_JITTER. [#7339]
 
+- Split NIRISS WFSS dual grism (gr150r+gr150c) associations into separate asn's for each grism. [#7351]
```
Comment (stscieisenhamer) [inline CHANGES.rst:22]: Going to need to rebase and fix since 1.8.4 has released.

**Review by stscieisenhamer — APPROVED** [review]
besides a rebase to fix the changelog, LGTM.

File: CHANGES.rst (hunk 1 above)
Comment (hbushouse) [reply CHANGES.rst:22]: fixed in latest commit

**Pull request merged**